core/src/main/java/com/linecorp/armeria/client/logging/LoggingClient.java

```diff
@@ -33,20 +36,22 @@
 public final class LoggingClient extends AbstractLoggingClient<HttpRequest, HttpResponse>
         implements HttpClient {
 
+    private static final Logger logger = LoggerFactory.getLogger(LoggingClient.class);
+
     /**
      * Returns a new {@link HttpClient} decorator that logs {@link Request}s and {@link Response}s at
      * {@link LogLevel#DEBUG} for success, {@link LogLevel#WARN} for failure. See {@link LoggingClientBuilder}
      * for more information on the default settings.
      */
     public static Function<? super HttpClient, LoggingClient> newDecorator() {
-        return builder().newDecorator();
+        return new LoggingClientBuilder().defaultLogger(logger).newDecorator();
```

Comment:
Should we use the `logger()` method instead of adding a new method? IIUC, the only difference is `setBuildLogWriter()`. It seems we need to remove `setBuildLogWriter()` in the `logger()` method. Because we can use a `LoggWriter.of(logger)` to create a `LogWriter` without a builder.
https://github.com/line/armeria/blob/9077eb482d15e8ab1d16ed25841af00d997880f8/core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java#L106-L106

```suggestion
        return new LoggingClientBuilder().logger(logger).newDecorator();
```

Reply:
> It seems we need to remove setBuildLogWriter() in the logger() method.

`LogWriter` and other properties are strictly mutually exclusive because a `LogWriter` implementation might not be relevant to a `logger`.
Let's say a `KafkaLogWriter` and a `logger`(which is not a default logger) are set together. We should raise an exception when the `logger` is set to the builder because it's not going to be used at all.